tests/test_columns.py

```diff
@@ -184,3 +184,55 @@ def test_select_column_from_same_table_multiple_time_using_different_alias():
     assert_column_lineage_equal(
         sql, [("tab2.col1", "tab1.col2"), ("tab2.col1", "tab1.col3")]
     )
+
+
+def test_comment_after_column():
+    sql = """INSERT OVERWRITE TABLE tab1
+    SELECT
+    a.col1
+    --, a.col2
+    , a.col3
+    FROM
+    tab2 a
+    """
+    assert_column_lineage_equal(
+        sql, [("tab2.col1", "tab1.col1"), ("tab2.col3", "tab1.col3")]
+    )
+
+
+def test_cast_with_comparison():
+    sql = """
+    INSERT OVERWRITE TABLE tab1
+    select cast(col1=1 as int) col1, col2=col3 col2
+    from tab2;
+    """
+    assert_column_lineage_equal(
+        sql, [("tab2.col1", "tab1.col1"), ("tab2.col2", "tab1.col2"), ("tab2.col3", "tab1.col2")]
+    )
+
+
+def test_complex_window_function():
```

Comment:
This is the only WIP part, right?

Reply:
Yes, and I decided to drop it from this PR because it clearly needs to be tackled in a different way